### ISSUE TITLE
A global variable should have the same digest value if it is uninitialized or zero initialized

### ISSUE THREAD

**Issue by MaggieYingYi** [SNSystems/llvm-prepo#22]
Compile the following code:
```
int a;          //uninitialized
int b = 0;   //Zero initialized

```
using
```
$ rm clang.db
$ clang -O0 -c -target x86_64-pc-linux-gnu-repo -o test.o test.c
```
This is being stored in the repository as:
```
pstore-dump -ticket test.o clang.db
- file    :
      path : clang.db
      size : 4194304
  tickets :
      - id     : 5e30ae41-98a7-421e-8eee-466de7bb9e4e
        ticket :
            members :
                ticket_member :
                    - digest  : 9094c1bc4813b45686785e1a0e2bc1ca
                      name    : a
                      linkage : external
                    - digest  : 5cd1a968567ef37fcd2aa2ba0cae58b2
                      name    : b
                      linkage : common
            path    : \\\\?\\C:\\MyWork\\repo_bug\\bug19

```

By default, 'a' is zero initialized. The digest of 'a' is wrong and should be the same as b.

**Comment by MaggieYingYi** [comment]
Fixed in git commit: 7d75c273be7619c.

**Comment by MaggieYingYi** [comment]
Fixed a unittes in git commit:6bd9f892218c95.
